components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/config/ConfigurationFacade.java

```diff
@@ -171,46 +173,81 @@ public ExternalIdPConfig getIdPConfigByRealm(String realm, String tenantDomain)
     }
 
     public String getAuthenticationEndpointURL() {
-        String authenticationEndpointURL = FileBasedConfigurationBuilder.getInstance().getAuthenticationEndpointURL();
-        if (StringUtils.isBlank(authenticationEndpointURL)) {
-            authenticationEndpointURL = "/authenticationendpoint/login.do";
+        String authenticationEndpointURL = "/authenticationendpoint/login.do";
```

Comment:
All the methods seem to have to the same logic,

1. Check tenant qualified and if so, build the tenant qualified URL
2. If not tenant qualified, build using the file-based config builder 

Just wondering whether we could extract the common login to a method that takes in functions as arguments

something like
buildUrl(String context, Function<String, String> legacyModeConfigFetchMethod)